### PR TITLE
Feature/redux store

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request: 
- Redux store create using Redux Toolkit.
- Created Logic to asynchronously fetch data and post data according to the backend API routes from the API through async functions created using createAsyncThunk module of reduxjs.toolkit.
- Populate the Redux store with created reducer slices to make it available to other React components.

### CODE REVIEW

File: src/redux/motorcylce/motorcylceSlice.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const fetchMotorcycles = createAsyncThunk(
+  'motorcycle/fetchMotorcycles',
+  async (_nill, { rejectWithValue }) => {
+    const retrieveMotorcycleAPI = 'http://localhost:3000/api/v1/motorcycles';
+    const requestOptions = {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        // Needs to be filled with Auth Header from sign in request
+        // Authorization: `Bearer ${process.env.REACT_APP_CARBON_API_KEY}`,
```
Comment (Beardless-sheik) [inline src/redux/motorcylce/motorcylceSlice.js:18]: Need to Add Auth Headers on our next iteration when Auth is completed

File: src/redux/reservation/reservationSlice.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  reservations: [],
+  isLoading: false,
+  error: false,
+};
+
+export const fetchReservations = createAsyncThunk(
+  'reservations/fetchReservations',
+  async (_nill, { rejectWithValue }) => {
+    const userId = '4'; // In our next iteration we need to get the
+    // userId from the JWT header as I believe its part of the payload
```
Comment (Beardless-sheik) [inline src/redux/reservation/reservationSlice.js:13]: User Id hardcoded but should be obtained from current signed in user

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import motorcycleReducer from './motorcylce/motorcylceSlice';
+import reservationReducer from './reservation/reservationSlice'
+
+export default configureStore({
+  reducer: {
+    motorcycle: motorcycleReducer,
+    reservation: reservationReducer,
+  },
+  // Middleware to show Redux Transactions: Please remove when we push to production
+  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
```
Comment (Beardless-sheik) [inline src/redux/store.js:12]: Middleware to show redux step-by-step transactions for debugging. Should be removed for production and only allowed to run if .env=DEVELOPMENT (future comment for deployment in the next few iterations)

**Review by roshan-bajgain — APPROVED** [review]
Hi @Beardless-sheik, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**